include/igl/readPLY.cpp

```diff
@@ -16,20 +17,20 @@ template <
   typename Ntype,
   typename UVtype>
 IGL_INLINE bool igl::readPLY(
-  const std::string filename,
+  const std::string & filename,
   std::vector<std::vector<Vtype> > & V,
   std::vector<std::vector<Ftype> > & F,
   std::vector<std::vector<Ntype> > & N,
   std::vector<std::vector<UVtype> >  & UV)
 {
-  using namespace std;
-  // Largely follows ply2iv.c
-  FILE * ply_file = fopen(filename.c_str(),"r");
-  if(ply_file == NULL)
-  {
+  try {
+    std::ifstream ply_stream(filename, std::ios::binary);
+    if (ply_stream.fail())
+          return false;
+    return readPLY(ply_stream,V,F,N,UV);
+  } catch (...) {
```

Comment:
Maybe we should catch `std::exception()` and display the `.what()` on `stderr`. Using `(...)` can catch stuff like `throw 2;` which is a bit weird.